src/js/components/stock-movement-wizard/modals/AddDestinationModal.jsx

```diff
@@ -67,64 +68,50 @@ const FIELDS = {
 const ADDRESS_FIELDS = {
   address: {
     type: TextField,
-    label: 'address.address.label',
+    label: 'react.locationsConfiguration.address.street.label',
     defaultMessage: 'Street address',
   },
   address2: {
     type: TextField,
-    label: 'address.address2.label',
+    label: 'react.locationsConfiguration.address.street2.label',
     defaultMessage: 'Street address 2',
   },
   city: {
     type: TextField,
-    label: 'address.city.label',
+    label: 'react.locationsConfiguration.address.city.label',
     defaultMessage: 'City',
   },
   stateOrProvince: {
     type: TextField,
-    label: 'address.stateOrProvince.label',
+    label: 'react.locationsConfiguration.address.stateOrProvince.label',
     defaultMessage: 'State/Province',
   },
   postalCode: {
     type: TextField,
-    label: 'address.postalCode.label',
+    label: 'react.locationsConfiguration.address.postalCode.label',
     defaultMessage: 'Postal code',
   },
   country: {
     type: TextField,
-    label: 'address.country.label',
+    label: 'react.locationsConfiguration.address.country.label',
     defaultMessage: 'Country',
   },
 };
 
 class AddDestinationModal extends Component {
   constructor(props) {
     super(props);
-    this.state = {
-      locationTypes: [],
-    };
-
-    this.fetchLocationTypes = this.fetchLocationTypes.bind(this);
+    this.mapToOptionsList = this.mapToOptionsList.bind(this);
   }
 
   componentDidMount() {
-    this.fetchLocationTypes();
+    this.props.fetchLocationTypes({ params: { activityCode: ActivityCode.DYNAMIC_CREATION } });
   }
-
-
-  fetchLocationTypes() {
-    const url = '/openboxes/api/locations/locationTypes?activityCode=DYNAMIC_CREATION';
-
-    apiClient.get(url)
-      .then((response) => {
-        const resp = response.data.data;
-        const locationTypes = _.map(resp, (locationType) => {
-          const [en, fr] = _.split(locationType.name, '|fr:');
-          return { ...locationType, label: this.props.locale === 'fr' && fr ? fr : en };
-        });
-
-        this.setState({ locationTypes });
-      });
+  mapToOptionsList(list) {
+    return _.map(list, (locationType) => {
+      const [en, fr] = _.split(locationType.name, '|fr:');
+      return { ...locationType, label: this.props.locale === 'fr' && fr ? fr : en };
+    });
```

Comment:
there is a util function to handle that:
https://github.com/openboxes/openboxes/blob/678dba1019dd668f9f6ec0db8684af51216f0b06/src/js/utils/translation-utils.jsx#L3-L8